colossus/src/main/scala/colossus/protocols/http/Header.scala

```diff
@@ -268,4 +278,82 @@ case class QueryParameters(parameters: Seq[(String, String)]) extends AnyVal{
 
 }
 
+class DateHeader(start: Long = System.currentTimeMillis) extends HttpHeader {
+  import java.util.Date
+  import java.text.SimpleDateFormat
+
+  private val formatter = new SimpleDateFormat(DateHeader.DATE_FORMAT)
+  
+  private def generate(time: Long) = HttpHeader("Date", formatter.format(new Date(time)))
+  private var lastDate = generate(start)
+  private var lastTime = start
+
+  def key = lastDate.key
+  def value = lastDate.value
+
+  def encoded: Array[Byte] = encoded(System.currentTimeMillis)
+
+  def encoded(time: Long): Array[Byte] = {
+    if (time >= lastTime + 1000) {
+      lastDate = generate(time)
+      lastTime = time
+    }
+    lastDate.encoded
+  }
+
+}
+
+object DateHeader {
+
+  val DATE_FORMAT = "EEE, MMM d yyyy HH:MM:ss z"
+
+}
+
+
+class HttpBody(private val body: Array[Byte], val contentType : Option[HttpHeader] = None)  {
+
+  def size = body.size
+
+  def encode(buffer: core.DataOutBuffer) {
+    if (size > 0) buffer.write(body)
+  }
+
+  def bytes: ByteString = ByteString(body)
+
+  override def equals(that: Any) = that match {
+    case that: HttpBody => that.bytes == this.bytes
+    case other => false
+  }
+
+  override def hashCode = body.hashCode
+
+  override def toString = bytes.utf8String
+
+}
+
+trait HttpBodyEncoder[T] {
+  def encode(data: T): HttpBody
+}
+
+trait HttpBodyEncoders {
+  implicit object ByteStringEncoder extends HttpBodyEncoder[ByteString] {
```

Comment:
I keep on finding reasons not to do it, but I still can't stop myself from suggesting that we should look into a way to have a JsonEncoder in here somehow.  Here is the wrong place, but just wanted to throw an idea out.  I can think of like 5 reasons why its bad right off the bat however :)